baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LambdaMethodReference.java

```diff
@@ -170,8 +168,8 @@ private Description convertMethodInvocations(
         }
 
         return buildFix(methodSymbol, methodInvocation, root, state, isLocal(methodInvocation))
-                .map(fix ->
-                        buildDescription(root).setMessage(MESSAGE).addFix(fix).build())
+                .filter(fix -> SuggestedFixes.compilesWithFix(fix, state))
```

Comment:
This is a quick and dirty fix, we can write something more precise in the future if we find performance suffers.

Reply:
Ahh, I didn't see a good way to do the ambiguous check on the lambda/method reference receiver, but this would catch it. Should we put this filter in the `buildFix` method itself to catch any other potential invalid transforms?

Reply:
I'm not aware of other paths that result in ambiguous behavior, I'd prefer to catch specific causes so we can add test cases for them, and hopefully write more precise fixes that don't involve recompilation.